_posts/2016-03-02-Lazily-Automation.markdown

```diff
@@ -0,0 +1,82 @@
+
+So I created [a Safari Extension](https://github.com/orta/github-clippers) that handles automatically deleting branches on specific repos for me. It's not a biggie, it's not fancy, it's under a [hundred lines of code](https://github.com/orta/github-clippers/blob/master/GitHub-Clippers.safariextension/github_clippers.js) - but it automates something that annoyed others about my behavior. That counts for something.
+
+### Commit Automation
+
+I write _reasonable_ commit messages, they're not [amazing](http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html), they're [not terrible](http://www.whatthecommit.com). They're just, [alright](https://github.com/artsy/energy/commits/master?author=orta). I wanted to start trying to hold everyone accountable for doing better, so I advocated for copying the person with the [best commit style](https://github.com/artsy/eigen/commits/master?author=alloy) on our team.
+
+Their style is to have commits in a format like `[Context] Thing I did.` - it is much better that `Thing I did.`. So I looked into how I could automate this, because I would very quickly forget to do this. Here's what I did:
+
+``` sh
+function branch() {
+  git checkout master;
+  git pull upstream master;
+  git checkout -b $1
+  git config branch.$1.description $2
+}
+
+function commit() {
+  local BRANCH=$(git_branch_info)
+  local INFO=$(git config branch.$(echo $BRANCH).description)
+  git commit -m "[$(echo $INFO)] $argv"
+}
+```
+
+I created two shell functions, one that makes a branch that includes a context type. So for example, say I'm working on artwork notifications, I'd start a new branch with `$ branch artwork_notifications Notifications`. This saves the context as `Notifications` on the git branch metadata. Then everytime I want to commit my changes, I use `$ commit This is the thing I changed.` - and it will be prefixed with `[Notifications]`. It makes it easier for someone looking through the history to have an idea about the context, and makes me feel like I'm improving my process without remembering the context.
+
+### Branch Prefixes
+
+We use a Makefile in all our projects to try and help automate per-project simple tasks like running [mogenertor](https://github.com/artsy/energy/blob/12fe9de4d927eea27f4942d15e74b89016a6345f/Makefile#L44-L47), updating [storyboard identifiers](https://github.com/artsy/energy/blob/12fe9de4d927eea27f4942d15e74b89016a6345f/Makefile#L49-L50) and updating [embedded resources](https://github.com/artsy/eigen/blob/12fe9de4d927eea27f4942d15e74b89016a6345f/Makefile#L102-L103).
+
+I also applied some standard make commands in our projects so that I can prefix my [branches with my name](https://github.com/artsy/eigen/blob/12fe9de4d927eea27f4942d15e74b89016a6345f/Makefile#L111-L118).
+
+``` sh
+LOCAL_BRANCH = $(shell git rev-parse --abbrev-ref HEAD)
+BRANCH = $(shell echo $(shell whoami)-$(shell git rev-parse --abbrev-ref HEAD))
```

Comment:
Use this instead, as `whoami`might not match github username https://github.com/artsy/eigen/commit/f46957aec8394133020447924490d739bb91314e